sig-security/joint-review.md

```diff
@@ -0,0 +1,375 @@
+
+Areas that are potentially susceptible to attacks would be credential handling, for example, token-based single sign-on from edge side to cloud side. However, all of these use some form of token based authentication and we assume the attacker cannot easily break existing cryptographic algorithms like AES or SHA256.
+
+### Attack Risks and Effects
+
+- Data saved in one database on the edge node, if an attacker attacks an edge database, it will cause denial of service in off-line scenarios.
+- External App can request Kubernetes resource to Meta-Server through list-watch method, then the request will be passed to CloudCore and finally to the api-server of the cluster. If an app was exploited by attacker, a DOS attack to list-watch server can be launched.
+
+### Security Degradation
+
+Exploiting vulnerabilities in management on the cloud can lead to maliciously deleting resources or bringing the whole cluster down. If an attacker intercepts tokens used for communicating with CloudCore, he can register a fake node and render the CloudCore unusable. If attackers gain access to the EdgeCore database, he could manipulate critical metadata such as adding or deleting existing pods or change the host mode or image pull address of pods.
+
+### Compensating Mechanisms
+
+- Please see the mitigations described at [here](https://github.com/kubeedge/community/blob/master/sig-security/sig-security-audit/KubeEdge-threat-model-and-security-protection-analysis.md#kubeedge-threat-model-and-mitigations), under the section `KubeEdge threat-model and mitigations`.
+
+## Threat Model
+
+Please see the detailed threat model analysis paper of KubeEdge at [here](https://github.com/kubeedge/community/blob/master/sig-security/sig-security-audit/KubeEdge-threat-model-and-security-protection-analysis.md).
+
+### Identity Theft
+
+Please refer to the above section `Threat Model`.
+
+### Compromise
+
+Please refer to the above section `Threat Model`.
+
+### Denial of Service
+
+Please refer to the above section `Threat Model`.
+
+## Secure Development Practices
+
+KubeEdge has achieved the passing level criteria for [CII Best Practices](https://bestpractices.coreinfrastructure.org/en/projects/3018).
+
+- Development Pipeline
+
+All code is maintained in [Git](https://github.com/kubeedge/kubeedge) and changes must be reviewed by maintainers and must pass all unit and e2e tests. Code changes are submitted via Pull Requests (PRs) and must be signed. Commits to the `main` branch directly are not allowed.

```

Comment:
becides UT and e2e, we have static checks, verifications on gofmt, go lint, go vet, vendors, and fossa check.